WarOfEmpires.QueryHandlers.Tests/Empires/GetWorkersQueryHandlerTests.cs

```diff
@@ -16,157 +13,58 @@
-        private readonly ResourcesMap _resourcesMap = new ResourcesMap();
-        private readonly EnumFormatter _formatter = new EnumFormatter();
-
-        public GetWorkersQueryHandlerTests() {
-            var user = Substitute.For<User>();
-            var player = Substitute.For<Player>();
-
-            user.Id.Returns(1);
-            user.Status.Returns(UserStatus.Active);
-            user.Email.Returns("test@test.com");
-
-            player.User.Returns(user);
-            player.Id.Returns(1);
-            player.Peasants.Returns(1);
-            player.Workers.Returns(new List<Workers>() {
-                new Workers(WorkerType.Farmers, 2),
-                new Workers(WorkerType.WoodWorkers, 3),
-                new Workers(WorkerType.StoneMasons, 4),
-                new Workers(WorkerType.OreMiners, 5),
-                new Workers(WorkerType.SiegeEngineers, 6),
-                new Workers(WorkerType.Merchants, 7),
-            });            
-            player.Tax.Returns(50);
-            player.GetRecruitsPerDay().Returns(5);
-            player.GetUpkeepPerTurn().Returns(new Resources(gold: 500, food: 30));
-            player.GetTotalResources().Returns(new Resources(gold: 1000, food: 100));
-            player.GetResourcesPerTurn().Returns(new Resources(gold: 400, food: 20));
-            player.GetBuildingBonus(BuildingType.SiegeFactory).Returns(4);
-            player.GetBuildingBonus(BuildingType.Market).Returns(10000);
-            player.HasUpkeepRunOut.Returns(true);
-
-            _context.Users.Add(user);
-            _context.Players.Add(player);
-        }
 
         [TestMethod]
-        public void GetWorkersQueryHandler_Returns_Correct_Peasants() {
+        public void GetWorkersQueryHandler_Returns_Correct_Information() {
+            var builder = new FakeBuilder().BuildPlayer(1)
+                .WithPeasants(1)
+                .WithWorkers(WorkerType.Farmers, 2)
+                .WithWorkers(WorkerType.WoodWorkers, 3)
+                .WithWorkers(WorkerType.StoneMasons, 4)
+                .WithWorkers(WorkerType.OreMiners, 5)
+                .WithWorkers(WorkerType.SiegeEngineers, 6)
+                .WithWorkers(WorkerType.Merchants, 7)
+                .WithBuilding(BuildingType.SiegeFactory, 4)
+                .WithBuilding(BuildingType.Market, 1);
+
+            builder.Player.Tax.Returns(50);
+            builder.Player.GetRecruitsPerDay().Returns(5);
+            builder.Player.GetUpkeepPerTurn().Returns(new Resources(gold: 500, food: 30));
+            builder.Player.GetTotalResources().Returns(new Resources(gold: 1000, food: 100));
+            builder.Player.GetResourcesPerTurn().Returns(new Resources(gold: 400, food: 20));
+            builder.Player.HasUpkeepRunOut.Returns(true);
+
+            var handler = new GetWorkersQueryHandler(builder.Context, new ResourcesMap(), new EnumFormatter());
             var query = new GetWorkersQuery("test@test.com");
```

Comment:
The FakeBuilder uses test{id}@test.com as default email address, and since you don't specify a different one when you create this player, this test doesn't find a matching email for me